### ISSUE TITLE
placeholder image not getting refreshed with newly added image for first time

### ISSUE THREAD

**Issue by pankajcoreview** [SDWebImage/SDWebImage#2681]
Hello Team,
I am using swift 4.3.  I have used below code.
`imageView.sd_setImage(with: imageUrl!, placeholderImage: #imageLiteral(resourceName: "PlaceholderSponsors"), options: .refreshCached, completed: nil)`

Previously url doesn't had an image hence image view was showing placeholder image which is correct behavior. 
But when I added image for the same URL and refreshed, it doesn't show updated image.
But if I killed the app and relaunched, its shows updated image.

**Comment by dreampiggy** [comment]
@pankajcoreview Please provide the issue template. Or we don't even know what SDWebImage/OS version you're talking about. And don't know what the `Previously` means for.

### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dreampiggy** [comment]
> Previously url doesn't had an image hence image view was showing placeholder image which is correct behavior.
> But when I added image for the same URL and refreshed, it doesn't show updated image.

Actually, can you explain this situation with more detail steps ? Seems I'm not get the idea of your operation, and the actual behavior.

Did you means this (I guess from your description):

1. I have a image server, and the loaded image url `URL1`. At the beginnning, the `URL1` does not associated to any image (the image server return something like 404 ?)
2. The image view showing the placeholder of `UIImage.named("xxx")`
3. Then I config my image server, to let `URL1` return a valid image URL (can sucessfully load from network)
4. But when I refresh the UI (Does not re-launch the App), the image still showing the placeholder of 
`UIImage.named("xxx")`
5. I think it's not correct. And I kill the App to re-launch. Howevcer, this time can successfully load the `URL1` image url.

If the above description it's what you face. The problem it's that you didn't specify a `.retryFailed` option in `SDWebImageOptions`. See the [documentation about it](https://sdwebimage.github.io/Enums/SDWebImageOptions.html#/c:@E@SDWebImageOptions@SDWebImageRetryFailed)

> By default, when a URL fail to be downloaded, the URL is blacklisted so the library won’t keep trying. This flag disable this blacklisting.

It's a historical behavior from SDWebImage 2.0, and it's the default behavior to block any error loaded image if you don't provide this options

(PS:Though I think it's a little aggressive to black list the URL **by default** than **by config**, but our users seems already adopted this behavior)

**Comment by pankajcoreview** [comment]
Hi dreampiggy,
Exactly this is what I was trying to explain. The solution that you provided,  perfectly worked for me.
Thank you so much.
